drivers/sensor/tdk/icp101xx/icp101xx_drv.c

```diff
@@ -0,0 +1,261 @@
+	case ICP101XX_MEAS_LOW_POWER_P_FIRST:
+		return 2;
+	case ICP101XX_MEAS_NORMAL_T_FIRST:
+	case ICP101XX_MEAS_NORMAL_P_FIRST:
+		return 7;
+	case ICP101XX_MEAS_LOW_NOISE_T_FIRST:
+	case ICP101XX_MEAS_LOW_NOISE_P_FIRST:
+		return 24;
+	default:
+	case ICP101XX_MEAS_ULTRA_LOW_NOISE_T_FIRST:
+	case ICP101XX_MEAS_ULTRA_LOW_NOISE_P_FIRST:
+		return 95;
+	}
+}
+
+static uint8_t get_conversion_ms(enum icp101xx_meas mode)
+{
+	switch (mode) {
+	case ICP101XX_MEAS_LOW_POWER_T_FIRST:
+	case ICP101XX_MEAS_LOW_POWER_P_FIRST:
+		return 1;
+	case ICP101XX_MEAS_NORMAL_T_FIRST:
+	case ICP101XX_MEAS_NORMAL_P_FIRST:
+		return 5;
+	case ICP101XX_MEAS_LOW_NOISE_T_FIRST:
+	case ICP101XX_MEAS_LOW_NOISE_P_FIRST:
+		return 20;
+	default:
+	case ICP101XX_MEAS_ULTRA_LOW_NOISE_T_FIRST:
+	case ICP101XX_MEAS_ULTRA_LOW_NOISE_P_FIRST:
+		return 80;
+	}
+}
+
+#ifdef ICP101XX_DRV_USE_FLOATS
+#define ATMOSPHERICAL_PRESSURE_KPA ((float)101.325)
+#define TO_KELVIN(temp_C)          ((float)273.15 + temp_C)
+/*
+ * Constant in altitude formula:
+ * M*g/R = (0,0289644 * 9,80665 / 8,31432)
+ * with M the molar mass of air.
+ * with g the gravitational force acceleration.
+ * with R the universal gaz constant.
+ */
+#define HEIGHT_TO_PRESSURE_COEFF   ((float)0.03424)
+
+/*
+ * Constant in altitude formula:
+ * R / (M*g) = 8,31432 / (0,0289644 * 9,80665)
+ * with M the molar mass of air.
+ * with g the gravitational force acceleration.
+ * with R the universal gaz constant.
+ */
+#define PRESSURE_TO_HEIGHT_COEFF   ((float)29.27127)
+/*
+ * Constant for altitude formula:
+ * logarithm of pressure at 0m
+ * ln(101.325)
+ */
+#define LOG_ATMOSPHERICAL_PRESSURE ((float)4.61833)
+
+float convertToHeight(float pressure_kp, float temperature_C)
+{
+	return PRESSURE_TO_HEIGHT_COEFF * TO_KELVIN(temperature_C) *
+	       (LOG_ATMOSPHERICAL_PRESSURE - logf(pressure_kp));
+}
+#endif
+
+static int icp101xx_attr_set(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, const struct sensor_value *val)
+{
+	int err = 0;
+	struct icp101xx_data *data = (struct icp101xx_data *)dev->data;
+
+	__ASSERT_NO_MSG(val != NULL);
+	if (chan == SENSOR_CHAN_PRESS) {
+		if (attr == SENSOR_ATTR_CONFIGURATION) {
+			if ((val->val1 >= ICP101XX_MEAS_LOW_POWER_T_FIRST) &&
+			    (val->val1 <= ICP101XX_MEAS_ULTRA_LOW_NOISE_P_FIRST)) {
+				data->icp_device.measurement_mode = val->val1;
+			} else {
+				LOG_ERR("Not supported ATTR value");
+				return -EINVAL;
+			}
+
+		} else {
+			LOG_ERR("Not supported ATTR");
+			return -EINVAL;
+		}
+	};
+	return err;
+}
+
+static int icp101xx_sample_fetch(const struct device *dev, const enum sensor_channel chan)
+{
+	struct icp101xx_data *data = (struct icp101xx_data *)dev->data;
+	int rc = 0;
+	uint64_t timeout;
+
+	if (!((chan == SENSOR_CHAN_AMBIENT_TEMP) || (chan == SENSOR_CHAN_PRESS) ||
+	      (chan == SENSOR_CHAN_ALTITUDE) || (chan == SENSOR_CHAN_ALL))) {
+		return -ENOTSUP;
+	}
+	rc = inv_icp101xx_enable_sensor(&data->icp_device, 1);
```

Comment:
Isn't this equivalent to i2c_write() with a simple register write? Why is the TDK HAL needed here?